workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -289,4 +322,17 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
       SideEffectNode(key, job)
     }
   }
+
+  @JvmInline
+  internal value class Box<T>(private val _value: Any? = Uninitialized) {
```

Comment:
Nit: This seems like a super generic and potentially useful helper, I'd pull it out.

Reply:
Extracted into `workflow-core` as `NullableInitBox` with its own tests.